services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,21 +46,56 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  BaseType_t checkQueueSend;
+  if (event == NULL) {
+    return ERR_CODE_INVALID_ARG;
+  }
+  checkQueueSend = xQueueSend(thermalMgrQueueHandle, (void *) event, (TickType_t) 10);
```

Comment:
when you pass in (TickType_t) 10, this means that this will block for 10 ticks if the queue is full, but you are calling this function from an ISR in osHandler so you should not be blocking